### ISSUE TITLE
Error is not instance of shaka.util.Error

### ISSUE THREAD

**Issue by aletorrado** [google/shaka-player#370]
This code is triggering an assert of this type:

```
  this.requestManifest_().then(function() {
    // Detect a call to stop()
    if (!this.networkingEngine_)
      return;

    // Ensure the next update occurs within |updatePeriod_| seconds by taking
    // into account the time it took to update the manifest.
    var endTime = Date.now();
    this.setUpdateTimer_((endTime - startTime) / 1000.0);
  }.bind(this)).catch(function(error) {
    goog.asserts.assert(error instanceof shaka.util.Error,
                        'Should only receive a Shaka error');
```

Don't know what the error is for (nothing gets printed), but here is the manifest:

```
<?xml version="1.0" encoding="utf-8"?>
<MPD xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xmlns="urn:mpeg:dash:schema:mpd:2011"
	xmlns:xlink="http://www.w3.org/1999/xlink"
	xsi:schemaLocation="urn:mpeg:DASH:schema:MPD:2011 http://standards.iso.org/ittf/PubliclyAvailableStandards/MPEG-DASH_schema_files/DASH-MPD.xsd"
	profiles="urn:mpeg:dash:profile:isoff-live:2011"
	type="dynamic"
	minimumUpdatePeriod="PT4S"
	suggestedPresentationDelay="PT4S"
	availabilityStartTime="2016-05-05T20:02:46"
	publishTime="2016-05-06T00:02:10"
	minBufferTime="PT4.1S">
	<ProgramInformation>
	</ProgramInformation>
	<Period start="PT0.0S">
		<AdaptationSet contentType="video" segmentAlignment="true" bitstreamSwitching="true" frameRate="30/1">
			<Representation id="0" mimeType="video/mp4" codecs="avc1.64001f" bandwidth="1200000" width="854" height="480" frameRate="30/1">
				<SegmentTemplate timescale="15360" initialization="init-stream$RepresentationID$.m4s" media="chunk-stream$RepresentationID$-$Number%05d$.m4s" startNumber="1">
					<SegmentTimeline>
						<S t="0" d="64000" r="1" />
						<S d="27136" />
						<!-- thousands of segments more -->
					</SegmentTimeline>
				</SegmentTemplate>
			</Representation>
			<Representation id="1" mimeType="video/mp4" codecs="avc1.64001f" bandwidth="2400000" width="1280" height="720" frameRate="30/1">
				<SegmentTemplate timescale="15360" initialization="init-stream$RepresentationID$.m4s" media="chunk-stream$RepresentationID$-$Number%05d$.m4s" startNumber="1">
					<SegmentTimeline>
						<S t="0" d="64000" />
						<S d="62464" />
						<!-- thousands of segments more -->
					</SegmentTimeline>
				</SegmentTemplate>
			</Representation>
		</AdaptationSet>
		<AdaptationSet contentType="audio" segmentAlignment="true" bitstreamSwitching="true">
			<Representation id="2" mimeType="audio/mp4" codecs="mp4a.40.2" bandwidth="96000" audioSamplingRate="44100">
				<AudioChannelConfiguration schemeIdUri="urn:mpeg:dash:23003:3:audio_channel_configuration:2011" value="2" />
				<SegmentTemplate timescale="44100" initialization="init-stream$RepresentationID$.m4s" media="chunk-stream$RepresentationID$-$Number%05d$.m4s" startNumber="1">
					<SegmentTimeline>
						<S t="0" d="181248" />
						<S d="179200" />
						<!-- thousands of segments more -->
					</SegmentTimeline>
				</SegmentTemplate>
			</Representation>
		</AdaptationSet>
	</Period>
</MPD>
```

Maybe it's the MPD size? (> 6000 segments)

**Comment by joeyparrish** [comment]
You should only see instances of shaka.util.Error at the application level.  The assertion means that there is a condition we are not handling properly.  Otherwise, the assertion itself is harmless.  There is, however, an error condition of some kind.  Since we are not reporting it correctly, it's hard to say what it is.  :-)

If you can provide a manifest URL, we can try it out and trace the real problem for you.

**Comment by aletorrado** [comment]
I cannot publish it yet, but I've just tested and `error===undefined`. It's strange.

**Comment by tdrews** [comment]
Thanks for the report. I can reproduce this issue. We'll take a look. Maybe related #369?

**Comment by tdrews** [comment]
Hi, since be09f52 I no longer see this problem as before. Are you still seeing this error from master?

**Comment by tdrews** [comment]
This seems to have been fixed by be09f52.
